lightning/src/ln/channelmonitor.rs

```diff
@@ -149,8 +155,8 @@ pub struct HTLCUpdate {
 }
 impl_writeable!(HTLCUpdate, 0, { payment_hash, payment_preimage, source });
 
-/// A simple implementation of a ManyChannelMonitor and ChainListener. Can be used to create a
-/// watchtower or watch our own channels.
+/// A simple implementation of a [`chain::Watch`]. Can be used to create a watchtower or to watch
```

Comment:
This seems like a strange place for this now - if Watch is in chain, why is the sample implementation in ln? Also likely worth noting that the watchtower stuff no longer makes sense in the comments here, which probably *also* implies we should drop the whole weird Key templating - the original thinking was that you'd Key on the OutPoint for local use and a (PublicKey, u32) for remote use where the originator signs updates with a PublicKey. Since ChannelMonitor is now never a watchtower, the Key differentiation makes no sense and we should probably go to just being a chain::Watch sample implementation.

Reply:
> Good question. I see it less as a sample implementation and more as a typically (re-)used component that also implements chain::Watch.

That doesn't really change the meat of the issue, though, no? If its really just a `chain::Watch` that you're supposed to use, why isn't it in `chain`, and why is it in `ln`?

> can I assume this outside the scope of this PR?

Sure, you can do it later, I don't feel strongly either way.

Reply:
> Also likely worth noting that the watchtower stuff no longer makes sense in the comments here, which probably _also_ implies we should drop the whole weird Key templating - the original thinking was that you'd Key on the OutPoint for local use and a (PublicKey, u32) for remote use where the originator signs updates with a PublicKey. Since ChannelMonitor is now never a watchtower, the Key differentiation makes no sense and we should probably go to just being a chain::Watch sample implementation.

Actually, I'll need some clarification to better understand the watchtower case. What was the `u32` meant to signify in the key?

Also, what is meant by "ChannelMonitor is now never a watchtower"? Did you mean `ChainMonitor`?

Reply:
No, up until (I believe) this year, ChannelMonitor contained an enum that allowed it to "be in watchtower mode" (ie not have private keys), but it was never fully implemented, so removed. In that context, it would make sense to store ChannelMonitors indexed by "public key of user who signed the update" and some channel id per user.